src/settings.ts

```diff
@@ -249,10 +249,17 @@ export var settings = {
   /**
    * Specifies environment in which SurveyJS will exist
    */
-  env: document,
 
-  get environment(): ISurveyEnvironment { return settings.env; },
-  set environment(environment: ISurveyEnvironment) { (settings.env as any) = environment; },
+  environment: {
+    root: document,
+    rootElement: document.body,
+    popupMountContainer: document.body,
+    svgMountContainer: document.head,
+    stylesSheetsMountContainer: document.head,
+    getElementById: document.getElementById,
+    addEventListener: document.addEventListener,
+    removeEventListener: document.removeEventListener
```

Comment:
I've double-checked it, and those 3 methods are necessary to override in the ShadowDOM env.

Without
```ts
settings.environment.getElementById = shadowRoot.getElementById.bind(shadowRoot)
```
elements inside the `shadowRoot` won't be found

Reply:
Hello,

I think, we can use environment.root.getElementById.

Reply:
That's right, good point 👍